org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/SynapseLanguageService.java

```diff
@@ -229,6 +229,7 @@ public void init(String projectUri, Object settings, SynapseLanguageClientAPI la
             log.log(Level.SEVERE, "Project path is null. Language server initialization failed.");
         }
         resourceFinder = ResourceFinderFactory.getResourceFinder(isLegacyProject);
+        resourceFinder.loadDependentResources(projectUri);
```

Comment:
Let's move this inside `if (projectUri != null) {` check.

Reply:
Fixed in 4c6eae30deb137da18cee7db8148e4cba20a1274